lib/src/posthog.dart

```diff
@@ -127,5 +127,52 @@ class Posthog {
 
   Future<String?> getSessionId() => _posthog.getSessionId();
 
+  /// Sets a callback to be invoked when feature flags are loaded from the PostHog server.
+  ///
+  /// The behavior of this callback differs slightly between platforms:
+  ///
+  /// **Web:**
+  /// The callback will receive:
+  /// - `flags`: A list of active feature flag keys (List<String>).
+  /// - `flagVariants`: A map of feature flag keys to their variant values (Map<String, dynamic>).
+  ///
+  /// **Mobile (Android/iOS):**
+  /// The callback serves primarily as a notification that the native PostHog SDK
+  /// has finished loading feature flags. In this case:
+  /// - `flags`: Will be an empty list.
+  /// - `flagVariants`: Will be an empty map.
+  /// After this callback is invoked, you can reliably use `Posthog().getFeatureFlag('your-flag-key')`
+  /// or `Posthog().isFeatureEnabled('your-flag-key')` to get the values of specific flags.
+  ///
+  /// **Common Parameters:**
+  /// - `errorsLoading` (optional named parameter): A boolean indicating if an error occurred during the request to load the feature flags.
+  ///   This is `true` if the request timed out or if there was an error. It will be `false` if the request was successful.
+  ///
+  /// This is particularly useful on the first app load to ensure flags are available before you try to access them.
+  ///
+  /// Example:
+  /// ```dart
+  /// Posthog().onFeatureFlags((flags, flagVariants, {errorsLoading}) {
+  ///   if (errorsLoading == true) {
+  ///     // Handle error, e.g. flags might be stale or unavailable
+  ///     print('Error loading feature flags!');
+  ///     return;
+  ///   }
+  ///   // On Web, you can iterate through flags and flagVariants directly.
+  ///   // On Mobile, flags and flagVariants will be empty here.
+  ///   // After this callback, you can safely query specific flags:
+  ///   final isNewFeatureEnabled = await Posthog().isFeatureEnabled('new-feature');
+  ///   if (isNewFeatureEnabled) {
+  ///     // Implement logic for 'new-feature'
+  ///   }
+  ///   final variantValue = await Posthog().getFeatureFlag('multivariate-flag');
+  ///   if (variantValue == 'test-variant') {
+  ///     // Implement logic for 'test-variant' of 'multivariate-flag'
+  ///   }
+  /// });
+  /// ```
+  void onFeatureFlags(OnFeatureFlagsCallback callback) =>
```

Comment:
Following the other SDKs, this should be a callback defined in the `PostHogConfig` during SDK init